spring-xd-module/src/main/java/org/springframework/xd/module/core/JavaConfiguredModule.java

```diff
@@ -62,8 +57,10 @@ protected void configureModuleApplicationContext(SimpleModuleDefinition moduleDe
 	}
 
 	public static String[] basePackages(SimpleModuleDefinition moduleDefinition, ClassLoader moduleClassLoader) {
-		Resource propertiesFile = ModuleUtils.locateModuleResource(moduleDefinition, moduleClassLoader,".properties");
-		Assert.notNull(propertiesFile, "required module properties not found.");
+		Resource propertiesFile = ModuleUtils.locateModuleResource(moduleDefinition, moduleClassLoader, ".properties");
+		if (propertiesFile == null) {
+			return new String[0];
```

Comment:
The reason for this is ModuleFactoryTests.missingConfigThrowsException(). It's hard to tell from the existing test, since it uses @Test(expected=..) and swallows the exception. The current implementation results in a better message:
java.lang.IllegalArgumentException: Required module artifacts are either missing or invalid. Unable to determine module type for module definition: 'processor:missing'.  Vs. "required module properties not found" from JavaConfiguredModule.basePackages(), which is misleading (what MP recently got prior to this change and found confusing).